src/content/docs/2/guide/plugins/develop-mobile.mdx

```diff
@@ -0,0 +1,394 @@
+  override fun load(webView: WebView) {
+    val timeout = this.config.getInt("timeout", 30)
+  }
+}
+```
+</TabItem>
+<TabItem label="iOS">
+
+```swift
+class ExamplePlugin: Plugin {
+  @objc public override func load(webview: WKWebView) {
+    let timeout = self.config["timeout"] as? Int ?? 30
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+## Lifecycle Events
+
+Plugins can hook into several lifecycle events:
+
+- [load](#load): When the plugin is loaded into the web view
+- [onNewIntent](#onnewintent): Android only, when the activity is re-launched
+
+There are also the additional [lifecycle events for plugins](/2/guide/plugins#lifecycle-events) in the Plugin Development guide.
+
+### load
+
+- **When**: When the plugin is loaded into the web view
+- **Why**: TODO:
+
+{/* TODO: above */}
+
+<Tabs>
+<TabItem label="Android">
+
+```java
+@TauriPlugin
+class ExamplePlugin(private val activity: Activity): Plugin(activity) {
+  override fun load(webView: WebView) {
+    // perform plugin setup here
+  }
+}
+```
+
+</TabItem>
+<TabItem label="iOS">
+
+```swift
+class ExamplePlugin: Plugin {
+  @objc public override func load(webview: WKWebView) {
+    let timeout = self.config["timeout"] as? Int ?? 30
+  }
+}
+```
+
+</TabItem>
+</Tabs>
+
+### onNewIntent
+
+**Note**: This is only available on Android.
+
+- **When**: When the activity is re-launched. See [Activity#onNewIntent](<https://developer.android.com/reference/android/app/Activity#onNewIntent(android.content.Intent)>) for more information.
+- **Why**: TODO:
+
+{/* TODO: above */}
+
+```java
+// import android.content.Intent
+
+@TauriPlugin
+class ExamplePlugin(private val activity: Activity): Plugin(activity) {
+  override fun onNewIntent(intent: Intent) {
+    // handle new intent event
+  }
+}
+```
+
+## Adding Mobile Commands
+
+There is a plugin class inside the respective mobile projects where commands can be defined that can be called by the Rust code:
+
+import { Tabs, TabItem } from '@astrojs/starlight/components';
+
+<Tabs>
+<TabItem label="Android">
+
+```java
+@TauriPlugin
+class ExamplePlugin(private val activity: Activity): Plugin(activity) {
+  @Command
+  fun openCamera(invoke: Invoke) {
+    val allowEdit = invoke.getBoolean("allowEdit", false)
+    val quality = invoke.getInt("quality", 100)
+
+    val ret = JSObject()
+    ret.put("path", "/path/to/photo.jpg")
+    invoke.resolve(ret)
+  }
+}
+```
+
+</TabItem>
+<TabItem label="iOS">
+
+
+```swift
+class ExamplePlugin: Plugin {
+	@objc public func openCamera(_ invoke: Invoke) {
+    let allowEdit = invoke.getBool("allowEdit", false)
+    let quality = invoke.getInt("quality", 100)
+
+    invoke.resolve(["path": "/path/to/photo.jpg"])
+	}
+}
+```
+
+</TabItem>
+</Tabs>
+
+Use the [`tauri::plugin::PluginHandle`](https://docs.rs/tauri/2.0.0-alpha/tauri/plugin/struct.PluginHandle.html) to call a mobile command from Rust:
+
+{/* TODO: Can this Rust code example be condensed down? */}
```

Comment:
If the below code can be simplified to really focus on what specifically needs covered for mobile plugins that would be amazing

Reply:
the code just defines an input parameter, output parameter, and how to call the mobile function

Reply:
yes Rust is verbose